File: backend/internal/model/mcp.go
package model

import "time"

type MCPConfig struct {
	BaseModel
	WorkspaceID   uint64     `gorm:"not null;index" json:"workspace_id"`
	Name          string     `gorm:"size:255;not null" json:"name"`
	Description   string     `gorm:"size:500" json:"description"`
	ServerURL     string     `gorm:"size:500;not null" json:"server_url"`
	TransportType string     `gorm:"size:20;default:sse" json:"transport_type"`
	APIKey        string     `gorm:"size:500" json:"api_key"`
	ToolsConfig   string     `gorm:"type:text" json:"tools_config"`
	IsEnabled     bool       `gorm:"default:true" json:"is_enabled"`
	LastSyncAt    *time.Time `json:"last_sync_at"`
	Workspace     Workspace  `gorm:"foreignKey:WorkspaceID" json:"-"`
}

func (MCPConfig) TableName() string {
	return "mcp_configs"
}

File: frontend/src/api/mcp.ts
/**
 * MCP API - MCP Server configuration and tool management
 */
import api from './index'

export interface MCPConfig {
  id: number
  name: string
  description: string
  workspace_id: number
  server_url: string
  transport_type: string
  is_enabled: boolean
  tools_count: number
  last_sync_at: string | null
  created_at: string
  updated_at: string
}

export interface MCPTool {
  name: string
  description: string
  input_schema: string
}

export interface MCPExecuteResult {
  [key: string]: any
}

export const mcpApi = {
  list: async (workspaceId: number): Promise<MCPConfig[]> => {
    const res = await api.get(`/workspaces/${workspaceId}/mcp`)
    return res.data
  },

  get: async (workspaceId: number, id: number): Promise<MCPConfig> => {
    const res = await api.get(`/workspaces/${workspaceId}/mcp/${id}`)
    return res.data
  },

  create: async (workspaceId: number, data: {
    name: string
    description?: string
    server_url: string
    transport_type?: string
    api_key?: string
    is_enabled?: boolean
  }): Promise<MCPConfig> => {
    const res = await api.post(`/workspaces/${workspaceId}/mcp`, data)
    return res.data
  },

  update: async (workspaceId: number, id: number, data: Partial<{
    name: string
    description: string
    server_url: string
    transport_type: string
    api_key: string
    is_enabled: boolean
  }>): Promise<MCPConfig> => {
    const res = await api.put(`/workspaces/${workspaceId}/mcp/${id}`, data)
    return res.data
  },

  delete: async (workspaceId: number, id: number): Promise<void> => {
    await api.delete(`/workspaces/${workspaceId}/mcp/${id}`)
  },

  discoverTools: async (workspaceId: number, id: number): Promise<MCPTool[]> => {
    const res = await api.post(`/workspaces/${workspaceId}/mcp/${id}/discover`)
    return res.data
  },

  getTools: async (workspaceId: number, id: number): Promise<MCPTool[]> => {
    const res = await api.get(`/workspaces/${workspaceId}/mcp/${id}/tools`)
    return res.data
  },

  executeTool: async (workspaceId: number, id: number, toolName: string, args: Record<string, any>): Promise<MCPExecuteResult> => {
    const res = await api.post(`/workspaces/${workspaceId}/mcp/${id}/execute`, {
      tool_name: toolName,
      arguments: args,
    })
    return res.data
  },
}
